src/components/search.js

```diff
@@ -188,7 +190,26 @@ class Search extends Emitter {
                     response.messages.reverse();
                 }
 
-                eachSeries(response.messages, this.triggerHistory, () => {
+                return new Promise((resolve, reject) => {
+
+                    let iteration = 0;
+                    let collection = response.messages;
+
+                    let forEachSeries = (iteratee, toRun) => {
```

Comment:
Is this really better? This was my train of thought:

* lets get rid of aysnc/eachSeries
* ok how do you do this in ES6
* I go to stack overflow
* That's a handy function
* This seems like a utility function that should be defined somewhere else
* Hrm, ok I should define this in another file
* Wait, isn't defining the utility function in another file conceptually the same as ```require('async/eachSeries')```

Reply:
I'm not saying don't do this, I'm saying that using a library over a custom utility may be a better bet and may not necessarily be larger.